src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -2654,6 +2656,61 @@ private Process StartWithShellExecute(ProcessStartInfo startInfo)
         #endregion
     }
 
+    /// <summary>
+    /// Provides argument completion for Verb parameter.
+    /// </summary>
+    public class VerbArgumentCompleter : IArgumentCompleter
+    {
+        private const string StartProcessCommandName = "Start-Process";
+        private const string FilePathParameterName = "FilePath";
+        private const string VerbParameterName = "Verb";
+
+        /// <summary>
+        /// Returns completion results for verb parameter.
+        /// </summary>
+        /// <param name="commandName">The command name.</param>
+        /// <param name="parameterName">The parameter name.</param>
+        /// <param name="wordToComplete">The word to complete.</param>
+        /// <param name="commandAst">The command AST.</param>
+        /// <param name="fakeBoundParameters">The fake bound parameters.</param>
+        /// <returns>List of Completion Results.</returns>
+        public IEnumerable<CompletionResult> CompleteArgument(
+            string commandName,
+            string parameterName,
+            string wordToComplete,
+            CommandAst commandAst,
+            IDictionary fakeBoundParameters)
+        {
+            // -Verb is not supported on non-Windows platforms as well as Windows headless SKUs
+            if (!Platform.IsWindowsDesktop
+                || !parameterName.Equals(VerbParameterName, StringComparison.OrdinalIgnoreCase))
```

Comment:
Why do we need the second check?

Reply:
We can probably remove it actually. This should always run in Verb parameter only.